test/apps/iodemo/io_demo.cc

```diff
@@ -628,7 +628,25 @@ class DemoServer : public P2pDemoCommon {
         listen_addr.sin_addr.s_addr = INADDR_ANY;
         listen_addr.sin_port        = htons(opts().port_num);
 
-        listen((const struct sockaddr*)&listen_addr, sizeof(listen_addr));
+        for (long i = 1;;) {
+            if (listen((const struct sockaddr*)&listen_addr,
+                       sizeof(listen_addr))) {
+                break;
+            }
+
+            UcxLog log(LOG_PREFIX);
+            log << "retry " << i;
+            if (opts().retries < std::numeric_limits<long>::max()) {
+                log << "/" << opts().retries;
+            }
+
+            if (++i <= opts().retries) {
+                log << " reconnecting in " << opts().retry_interval << " seconds";
```

Comment:
overall log message should be
"restarting listener on 0.0.0.0:2000 in 10 seconds (retry 1/5)"
...
"restarting listener on 0.0.0.0:2000 in 10 seconds (retry 5/5)"